java/spark/src/main/java/com/logicalclocks/hsfs/spark/engine/hudi/HudiEngine.java

```diff
@@ -407,10 +395,6 @@ public void streamToHoodieTable(SparkSession sparkSession, StreamFeatureGroup st
       hudiWriteOpts.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
```

Comment:
Is this condition still valid in the single kafka topic case?

Reply:
yes, after single Kafka topic the initial offset should almost always be provided to the Hudi engine. However, when it was not provided or previously check pointed in commit file then this conditions plays a role, forcing the job to start from the earliest offset.

`AUTO_OFFSET_RESET_CONFIG defines how a consumer should behave when consuming from a topic partition when there is no initial offset`